src/java/org/apache/cassandra/io/sstable/Descriptor.java

```diff
@@ -260,14 +263,14 @@ public static Descriptor fromFilename(File file)
         if (!Version.validate(versionString))
             throw invalidSSTable(name, "invalid version %s", versionString);
 
-        int generation;
+        SSTableUniqueIdentifier generation;
         try
         {
-            generation = Integer.parseInt(tokens.get(1));
+            generation = SSTableUniqueIdentifierFactory.instance.fromString(tokens.get(1));
         }
-        catch (NumberFormatException e)
+        catch (RuntimeException e)
```

Comment:
Wouldn't it be `IllegalArgumentException`?

Reply:
well, there is no way to enforce that IAE will be the only possible exception thrown by that method. I think we should interpret all runtime exceptions the same way and provide the appropriate message when re-throwing

Reply:
Should we do the same with the try-catch with `IllegalArgumentException` for the format part of the name, a few lines below?